lib/src/signed_video_h26x_auth.c

```diff
@@ -691,6 +691,39 @@ compute_gop_hash(signed_video_t *self, h26x_nalu_list_item_t *sei)
   return status;
 }
 
+/**
+ * Decodes the SEI message, retrieves necessary parameters for authentication, and computes the hash
+ * for authenticity.
+ */
+static svrc_t
+prepare_golden_sei(signed_video_t *self, h26x_nalu_list_item_t *sei)
+{
+  assert(self);
+  svrc_t status = SV_UNKNOWN_FAILURE;
+  sign_or_verify_data_t *verify_data = self->verify_data;
+  const size_t hash_size = verify_data->hash_size;
+  self->gop_state.has_sei = true;
+  SV_TRY()
+    // If an SEI item is found and it hasn't been decoded yet, decode it.
+    if (sei && !sei->has_been_decoded) {
+      // Extract the TLV data and size from the NALU.
+      const uint8_t *tlv_data = sei->nalu->tlv_data;
+      size_t tlv_size = sei->nalu->tlv_size;
+
+      // Decode the SEI data and update the status.
+      SV_THROW(decode_sei_data(self, tlv_data, tlv_size));
+      sei->has_been_decoded = true;  // Mark the SEI as decoded.
+      // Assuming the signature hash type is always DOCUMENT_HASH.
+      SV_THROW(hash_and_add_for_auth(self, sei));
+      memcpy(verify_data->hash, sei->hash, hash_size);
```

Comment:
I think using `verify_data->hash_size` fits on this line, so you can remove the declaration on line 704, since it is only used here.